Tests/PackageLoadingTests/PDLoadingTests.swift

```diff
@@ -30,30 +30,30 @@ class PackageDescriptionLoadingTests: XCTestCase, ManifestLoaderDelegate {
         fatalError("implement in subclass")
     }
 
-    func loadManifest(
+    func loadAndValidateManifest(
         _ contents: String,
         toolsVersion: ToolsVersion? = nil,
         packageKind: PackageReference.Kind? = nil,
         observabilityScope: ObservabilityScope,
         file: StaticString = #file,
         line: UInt = #line
-    ) throws -> Manifest {
-        try self.loadManifest(
+    ) throws -> (manifest: Manifest, diagnostics: [Basics.Diagnostic])  {
```

Comment:
the validator does not take an observability scope, instead it returns an array of diagnostics which the caller can decide what to with

Reply:
What's the advantage of doing it that way rather than a custom scope that just captures them and then passes them on?  That would allow the validation to use the normal pattern for emitting diagnostics.

Reply:
an API design preference: a validator is designed to return a list of validation issues vs. some functional api that _may_ need to emit diagnostics so takes a "scope" to do so